akiuri/akiuri.go

```diff
@@ -12,19 +19,26 @@ const (
 type ObjectType int
 
 const (
-	UNKNOWN_TYPE ObjectType = iota
-	SPEC
-	TRACE // aka learn session
+	SPEC  ObjectType = iota
+	TRACE            // aka learn session
 )
 
-func stringToObjectType(s string) ObjectType {
+func (o ObjectType) Ptr() *ObjectType {
+	return &o
+}
```

Comment:
I like everything about this but .Ptr() but I can't come up with an alternative I like more, so I guess we're going with it.

What I was thinking about: 

1. export ObjectType as a pointer type instead of int? I don't think we can have consts that are addresses.

2. export an interface type

Reply:
Perhaps an explicit `NONE` object type, rather than using `nil` to represent it?  Then there's no need to use pointers at all.

That being said, I think this is one of those things where this implementation gets the job done 🙂 .  I'm happy with it going in as is.

Reply:
I thought about that, but then the implementation of `String()` becomes janky. Not clear what to return there. The current implementation side-steps the issue by requiring an `ObjectType` instance before you can call `String()`.

Reply:
I also considered using option types in the client code, but apparently, those aren't really a thing in Go.